### ISSUE TITLE
allow numpy calls in functions in configs

### ISSUE THREAD

**Issue by esheldon** [GalSim-developers/GalSim#776]
Allow either `numpy.` or `np.` calls, for example the following.  I know this works in other scenarios such as Eval so should be straightforward to implement

```
        type: GBeta
        g:
            # this didn't work, due to np.exp
            type: RandomDistribution
            function: "(1-x**2)**2 * np.exp( -0.5 * x**2 / 0.2**2 )"
            x_min: 0.0
            x_max: 1.0
        beta:
            type: Random

```

**Comment by rmjarvis** [comment]
They are allowed.  You just have to write `numpy`, not `np`.  `math` will also work, since `x` isn't an array.  As documented [here](https://github.com/GalSim-developers/GalSim/wiki/Config-Values#preset-variables), the complete list of modules that will be imported for you prior to the `eval` command are:
```
numpy
math
os
galsim
```

**Comment by esheldon** [comment]
OK, thanks. I think I was confused because this example used `np` instead of `numpy` in the `e_power_function`

https://github.com/GalSim-developers/GalSim/blob/releases/1.3/examples/lsst.yaml#L129

**Comment by rmjarvis** [comment]
Ah.  That's probably an error actually.

OTOH, we could certainly do both `numpy` and `np`, since the latter is of course quite a common alias.  So it would probably be less confusing to just allow both.

**Comment by rmjarvis** [comment]
>  we could certainly do both numpy and np

This is done on master.  552647c9cbf2836ada902d52fcb6daad625c4c49

I don't think there is any other action item here, so I'll close this now.

**Comment by esheldon** [comment]
Using current master only `np` works, both `numpy` and `math` fail with errors

```
ValueError: String function must either be a valid filename or something that can eval to a function of x. Input provided: x*(1.0 - x**2)**2 * math.exp( -0.5*x**2 / 0.2**2 )
```

```
ValueError: String function must either be a valid filename or something that can eval to a function of x. Input provided: x*(1.0 - x**2)**2 * numpy.exp( -0.5*x**2 / 0.2**2 )
```

**Comment by rmjarvis** [comment]
You're right.  I actually misunderstood this issue when you first posted it.  This isn't using the config Eval functionality, but rather the DistDeviate evaluation of a string, which is a different bit of code.  That does indeed require `np` currently (and not `numpy` anymore).  I'll add a test of this snippet to make sure we allow both (and math as well, which I think never worked here).

**Comment by rmjarvis** [comment]
I fixed this on branch #776.  I want to add some more unit tests here, since there are other places where we eval user input that should have numpy and math.  I fixed the code there too, but I didn't write tests for them yet.

**Comment by rmjarvis** [comment]
Closed via PR #796.
